packages/features/ee/video/ViewRecordingsDialog.tsx

```diff
@@ -40,116 +41,155 @@ interface GetTimeSpanProps {
+  const [recordingId, setRecordingId] = useState("");
+  const { isFetching, data } = trpc.viewer.getDownloadLinkOfCalVideoRecordings.useQuery(
+    {
+      recordingId,
+    },
+    {
+      enabled: !!recordingId,
+      cacheTime: 0,
+      refetchOnWindowFocus: false,
+      refetchOnReconnect: false,
+      retry: false,
+      onSuccess: (data) => {
+        if (data && data.download_link) {
+          window.location.href = data.download_link;
+        }
+      },
+    }
+  );
+
+  return {
+    setRecordingId: (newRecordingId: string) => {
+      // may be a way to do this by default, but this is easy enough.
+      if (recordingId === newRecordingId && data) {
+        window.location.href = data.download_link;
+      }
+      if (!isFetching) {
+        setRecordingId(newRecordingId);
+      }
+      // assume it is still fetching, do nothing.
+    },
+    isFetching,
+  };
+};
+
+const ViewRecordingsList = ({ roomName, hasPaidPlan }: { roomName: string; hasPaidPlan: boolean }) => {
+  const { t } = useLocale();
+  const { setRecordingId, isFetching } = useRecordingDownload();
+
+  const { data: recordings } = trpc.viewer.getCalVideoRecordings.useQuery(
+    { roomName },
+    {
+      suspense: true,
+    }
+  );
+
+  const handleDownloadClick = async (recordingId: string) => {
+    // this would enable the getDownloadLinkOfCalVideoRecordings
+    setRecordingId(recordingId);
+  };
+
+  return (
+    <>
+      {recordings && "data" in recordings && recordings?.data?.length > 0 ? (
+        <div className="flex flex-col gap-3">
+          {recordings.data.map((recording: RecordingItemSchema, index: number) => {
+            return (
+              <div
+                className="flex w-full items-center justify-between rounded-md border px-4 py-2"
+                key={recording.id}>
+                <div className="flex flex-col">
+                  <h1 className="text-sm font-semibold">
+                    {t("recording")} {index + 1}
+                  </h1>
+                  <p className="text-sm font-normal text-gray-500">
+                    {convertSecondsToMs(recording.duration)}
+                  </p>
+                </div>
+                {hasPaidPlan ? (
+                  <Button
+                    StartIcon={FiDownload}
+                    className="ml-4 lg:ml-0"
+                    loading={isFetching}
+                    onClick={() => handleDownloadClick(recording.id)}>
+                    {t("download")}
+                  </Button>
+                ) : (
+                  <UpgradeTeamsBadge />
+                )}
+              </div>
+            );
+          })}
+        </div>
+      ) : (
+        (!recordings || (recordings && "total_count" in recordings && recordings?.total_count === 0)) && (
+          <p className="font-semibold">{t("no_recordings_found")}</p>
+        )
+      )}
+    </>
+  );
+};
+
 export const ViewRecordingsDialog = (props: IViewRecordingsDialog) => {
```

Comment:
Decreasing the amount of rerenders of this component improves overal performance - setIsOpenDialog rerenders the parent too and all children.